### PR TITLE
mcrl2: 201707 -> 202006

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
I noticed this package was [failing on hydra](https://hydra.nixos.org/build/133435077#tabs-summary), so I tried to update it.

It currently builds successfully, but the Qt GUI programs don't launch correctly.

```
$ /nix/store/3i1n3zvx5ibdp9s2f75aq44cklqqg0zw-mcrl2-202006/bin/mcrl2-gui 
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```

This is probably a simple fix with some Qt wrapper, but this is not an area of NixOS that I'm familiar with yet.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rmcgibbo** [comment]
I think I figured out the Qt fix. GUI launches now.

File: pkgs/applications/science/logic/mcrl2/default.nix [hunk 1]
```diff
@@ -1,19 +1,20 @@
-{stdenv, fetchurl, cmake, libGLU, libGL, qt5, boost}:
+{stdenv, fetchurl, cmake, libGLU, libGL, qt5, boost }:
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/mcrl2/default.nix:1]:
```suggestion
{ stdenv, fetchurl, cmake, libGLU, libGL, qt5, boost }:
```

File: pkgs/applications/science/logic/mcrl2/default.nix [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.0.tar.gz";
+    sha256 = "167ryrzk1a2j53c2j198jlxa98amcaym070gkcj730619gymv5zl";
   };
 
+  nativeBuildInputs = [ qt5.wrapQtAppsHook ];
   buildInputs = [ cmake libGLU libGL qt5.qtbase boost ];
 
   enableParallelBuilding = true;
 
+
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/mcrl2/default.nix:17]:
```suggestion
```

File: pkgs/applications/science/logic/mcrl2/default.nix [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
-    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.${build_nr}.tar.gz";
-    sha256 = "1c8h94ja7271ph61zrcgnjgblxppld6v22f7f900prjgzbcfy14m";
+    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.0.tar.gz";
+    sha256 = "167ryrzk1a2j53c2j198jlxa98amcaym070gkcj730619gymv5zl";
   };
 
+  nativeBuildInputs = [ qt5.wrapQtAppsHook ];
   buildInputs = [ cmake libGLU libGL qt5.qtbase boost ];
 
   enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/mcrl2/default.nix:15]:
```suggestion
```

Set by cmake.

File: pkgs/applications/science/logic/mcrl2/default.nix [hunk 4]
```diff
@@ -1,19 +1,20 @@
-{stdenv, fetchurl, cmake, libGLU, libGL, qt5, boost}:
+{stdenv, fetchurl, cmake, libGLU, libGL, qt5, boost }:
 
 stdenv.mkDerivation rec {
-  version = "201707";
-  build_nr = "1";
+  version = "202006";
   pname = "mcrl2";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/mcrl2/default.nix:5]:
```suggestion
  pname = "mcrl2";
  version = "202006.0";
```

File: pkgs/applications/science/logic/mcrl2/default.nix [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
 
   src = fetchurl {
-    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.${build_nr}.tar.gz";
-    sha256 = "1c8h94ja7271ph61zrcgnjgblxppld6v22f7f900prjgzbcfy14m";
+    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.0.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/logic/mcrl2/default.nix:8]:
```suggestion
    url = "https://www.mcrl2.org/download/release/mcrl2-${version}.tar.gz";
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107441` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>mcrl2</li>
  </ul>
</details>

Cmake configure fails.

**Comment by rmcgibbo** [comment]
Here's the  CMake configuration failure. IIUC it configures fine with CMake 3.18, but fails with CMake 3.19.

Marking as draft.

```
these derivations will be built:
  /nix/store/qdn74vf65jpzi8w35sj49zx3p1j0viq7-mcrl2-202006.0.drv
  /nix/store/02fbb0l98w0khrlglpqg3d5c2s5jamsc-env.drv
building '/nix/store/qdn74vf65jpzi8w35sj49zx3p1j0viq7-mcrl2-202006.0.drv'...
unpacking sources
unpacking source archive /nix/store/jwiyac3s8ghaw5yap8m42ai656wc1zrs-mcrl2-202006.0.tar.gz
source root is mcrl2-202006.0
setting SOURCE_DATE_EPOCH to timestamp 1594994227 of file mcrl2-202006.0/build/SourceVersion
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/share/doc/mCRL2 -DCMAKE_INSTALL_INFODIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/kcp0y7g1ir4dxq2gqz687i4vb7gqy8s0-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/kcp0y7g1ir4dxq2gqz687i4vb7gqy8s0-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/kcp0y7g1ir4dxq2gqz687i4vb7gqy8s0-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/b96dqbx6pri2xp2xxlq6i269virrdaw6-gcc-wrapper-9.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at 3rd-party/tr/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Performing Test C_ACCEPTS_STD_C99
-- Performing Test C_ACCEPTS_STD_C99 - Success
-- Performing Test C_ACCEPTS_WALL
-- Performing Test C_ACCEPTS_WALL - Success
-- Performing Test C_ACCEPTS_WNO_INLINE
-- Performing Test C_ACCEPTS_WNO_INLINE - Success
-- Performing Test C_ACCEPTS_FNO_STRICT_OVERFLOW
-- Performing Test C_ACCEPTS_FNO_STRICT_OVERFLOW - Success
-- Performing Test C_ACCEPTS_PIPE
-- Performing Test C_ACCEPTS_PIPE - Success
-- Performing Test C_ACCEPTS_PEDANTIC
-- Performing Test C_ACCEPTS_PEDANTIC - Success
-- Performing Test C_ACCEPTS_W
-- Performing Test C_ACCEPTS_W - Success
-- Performing Test C_ACCEPTS_FPROFILE_ARCS
-- Performing Test C_ACCEPTS_FPROFILE_ARCS - Failed
-- Performing Test C_ACCEPTS_FTEST_COVERAGE
-- Performing Test C_ACCEPTS_FTEST_COVERAGE - Success
-- Performing Test C_ACCEPTS_FPIC
-- Performing Test C_ACCEPTS_FPIC - Success
-- Performing Test CXX_ACCEPTS_STD_CPP17
-- Performing Test CXX_ACCEPTS_STD_CPP17 - Success
-- Performing Test CXX_ACCEPTS_WALL
-- Performing Test CXX_ACCEPTS_WALL - Success
-- Performing Test CXX_ACCEPTS_WNO_INLINE
-- Performing Test CXX_ACCEPTS_WNO_INLINE - Success
-- Performing Test CXX_ACCEPTS_FNO_STRICT_OVERFLOW
-- Performing Test CXX_ACCEPTS_FNO_STRICT_OVERFLOW - Success
-- Performing Test CXX_ACCEPTS_PIPE
-- Performing Test CXX_ACCEPTS_PIPE - Success
-- Performing Test CXX_ACCEPTS_PEDANTIC
-- Performing Test CXX_ACCEPTS_PEDANTIC - Success
-- Performing Test CXX_ACCEPTS_W
-- Performing Test CXX_ACCEPTS_W - Success
-- Performing Test CXX_ACCEPTS_WEXTRA
-- Performing Test CXX_ACCEPTS_WEXTRA - Success
-- Performing Test CXX_ACCEPTS_WUNUSED_VARIABLE
-- Performing Test CXX_ACCEPTS_WUNUSED_VARIABLE - Success
-- Performing Test CXX_ACCEPTS_WUNUSED_PARAMETER
-- Performing Test CXX_ACCEPTS_WUNUSED_PARAMETER - Success
-- Performing Test CXX_ACCEPTS_WUNUSED_FUNCTION
-- Performing Test CXX_ACCEPTS_WUNUSED_FUNCTION - Success
-- Performing Test CXX_ACCEPTS_WUNUSED
-- Performing Test CXX_ACCEPTS_WUNUSED - Success
-- Performing Test CXX_ACCEPTS_WNO_SYSTEM_HEADERS
-- Performing Test CXX_ACCEPTS_WNO_SYSTEM_HEADERS - Success
-- Performing Test CXX_ACCEPTS_WOVERLOADED_VIRTUAL
-- Performing Test CXX_ACCEPTS_WOVERLOADED_VIRTUAL - Success
-- Performing Test CXX_ACCEPTS_WWRITE_STRINGS
-- Performing Test CXX_ACCEPTS_WWRITE_STRINGS - Success
-- Performing Test CXX_ACCEPTS_WMISSING_DECLARATIONS
-- Performing Test CXX_ACCEPTS_WMISSING_DECLARATIONS - Success
-- Performing Test CXX_ACCEPTS_FPROFILE_ARCS
-- Performing Test CXX_ACCEPTS_FPROFILE_ARCS - Failed
-- Performing Test CXX_ACCEPTS_FTEST_COVERAGE
-- Performing Test CXX_ACCEPTS_FTEST_COVERAGE - Success
-- Performing Test CXX_ACCEPTS_FPIC
-- Performing Test CXX_ACCEPTS_FPIC - Success
CMake Warning at build/cmake/MCRL2Version.cmake:80 (message):
  No version information could be included because /build/mcrl2-202006.0 is
  unversioned.
Call Stack (most recent call first):
  CMakeLists.txt:82 (include)


CMake Deprecation Warning at 3rd-party/dparser/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/atermpp/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/bes/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Error at /nix/store/9wdpj1db32jd7xdqnxs0jk3cz6wc336y-cmake-3.19.1/share/cmake-3.19/Modules/Internal/CheckCompilerFlag.cmake:46 (message):
  check_compiler_flag: CXX: unknown language.
Call Stack (most recent call first):
  /nix/store/9wdpj1db32jd7xdqnxs0jk3cz6wc336y-cmake-3.19.1/share/cmake-3.19/Modules/CheckCXXCompilerFlag.cmake:40 (cmake_check_compiler_flag)
  libraries/data/CompilingRewriter.cmake:28 (check_cxx_compiler_flag)
  libraries/data/CMakeLists.txt:4 (include)


CMake Error at /nix/store/9wdpj1db32jd7xdqnxs0jk3cz6wc336y-cmake-3.19.1/share/cmake-3.19/Modules/Internal/CheckCompilerFlag.cmake:46 (message):
  check_compiler_flag: CXX: unknown language.
Call Stack (most recent call first):
  /nix/store/9wdpj1db32jd7xdqnxs0jk3cz6wc336y-cmake-3.19.1/share/cmake-3.19/Modules/CheckCXXCompilerFlag.cmake:40 (cmake_check_compiler_flag)
  libraries/data/CompilingRewriter.cmake:34 (check_cxx_compiler_flag)
  libraries/data/CMakeLists.txt:4 (include)


CMake Deprecation Warning at libraries/data/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/lps/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/pbes/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/pg/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/smt/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at libraries/utilities/example/CMakeLists.txt:11 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- **
-- ** Building mCRL2 202006.0.Unknown Release build (Shared)
-- **
-- ** Installing to:   /nix/store/lawxwnivfaxwc7czisy1wxvwvn0bq7p2-mcrl2-202006.0
-- ** CMake version:   3.19.1
-- ** C++ compiler:    GNU 9.3.0
-- ** C compiler:      GNU 9.3.0
-- ** Size of void*:   8
-- **
-- ** Boost:  1.69.
-- ** Qt:     QtCore 5.15.0, QtOpenGL 5.15.0, QtWidgets 5.15.0, QtGui 5.15.0, QtXml 5.15.0.
-- **
-- Configuring incomplete, errors occurred!
See also "/build/mcrl2-202006.0/build/CMakeFiles/CMakeOutput.log".
See also "/build/mcrl2-202006.0/build/CMakeFiles/CMakeError.log".
note: keeping build directory '/tmp/nix-build-mcrl2-202006.0.drv-2'
builder for '/nix/store/qdn74vf65jpzi8w35sj49zx3p1j0viq7-mcrl2-202006.0.drv' failed with exit code 1
cannot build derivation '/nix/store/02fbb0l98w0khrlglpqg3d5c2s5jamsc-env.drv': 1 dependencies couldn't be built
error: build of '/nix/store/02fbb0l98w0khrlglpqg3d5c2s5jamsc-env.drv' failed
```

**Comment by rmcgibbo** [comment]
It could be this issue, but I'm not qualified to say: https://gitlab.kitware.com/cmake/cmake/-/issues/21500

**Pull request closed without merging**